File: project_euler/4/main.cpp
#include<iostream>
using namespace std;

bool is_pali(int in){
    int j = 0;
    int i = in;
    while ( i ){
        j *= 10;
        j += i % 10;
        i = i / 10;
    }
    return in == j;

}

int main(){
    int res = 0;
    for(int i = 100; i < 1000; i++){
        for(int j = i + 1; j < 1000; j++){
            if(is_pali(i*j) && (i * j) > res){
                res = i * j;
            }
        }
    }
    cout << res << endl;
}

File: project_euler/4/main.py


def is_pali(number):
    temp1 = number
    temp2 = 0
    while(temp1):
        temp2 *= 10
        temp2 += temp1 % 10
        temp1 /= 10;
    return temp2 == number

res = 0

for a in range(100,999):
    for b in range(a + 1,999):
        if(is_pali(a*b) and a*b > res):
            res = a*b

print(res)
